File: scripts/helpers/does-file-exist.js
import { stat } from 'fs/promises';

/**
 *
 * @param {import('fs').PathLike} filePath
 * @returns {Promise<boolean>}
 */
export default async function doesFileExist(filePath) {
  try {
    await stat(filePath);
    return true;
  } catch (error) {
    if (error.code === 'ENOENT') {
      return false;
    }
    throw error;
  }
}

File: revamp/preprocessors/src/helpers/does_file_exist.go
package helpers

import (
	"log"
	"os"
)

func DoesFileExist(filePath string) bool {
	_, err := os.Stat(filePath)
	if os.IsNotExist(err) {
		return false
	} else if err != nil {
		log.Fatal(err)
	}
	return true
}
